src/web/lib/components/ThemeCustomBackgroundPicker/index.js

```diff
@@ -327,12 +326,19 @@ class ThemeCustomBackgroundSelector extends React.Component {
 
   handleClearBackground = () => {
     if (localStorage.getItem("clearImageModal")) {
-      this.props.clearCustomBackground();
+      this.removeImage();
     } else {
       this.props.setDisplayRemoveImageModal({ display: true });
     }
   };
 
+  removeImage = () => {
+    // TODO: update / remove /resave theme/s without this image.
+    const { image } = this.props;
+    this.props.clearCustomBackground();
+    this.props.storage.imageStorage.delete(image.name);
```

Comment:
The image is only present because it is in one of the entries in the `themePast` array at https://github.com/mozilla/FirefoxColor/blob/17092506ea355f49894eafe406a675cc114af9f3/src/web/lib/middleware.js#L86

This was a consciously added change in #418, as part of a fix to fix #414.
When we are not interested in the image, then we might as well remove the code from `purgeImages` in middleware.js, starting from `// Scan through undo/redo stack for images still in use.` until the end, i.e. https://github.com/mozilla/FirefoxColor/blob/17092506ea355f49894eafe406a675cc114af9f3/src/web/lib/middleware.js#L85-L96

Could you check whether that would also fix your bug and still not regress on #414 ?

Reply:
This is what we want: we want to remove the background image from local storage when the user clicks the X here (the user is warned about this the first time they try to do this. this was the purpose of #852). so yes it looks like I can remove this logic: `// Scan through undo/redo stack for images still in use...`  
Regarding “undo/redo”, I tried to change this with #854 but unfortunately I do still see a bug here. So I can try to remove this action from the undo/redo action list or maybe set up another action to flag this but I’ll have to look into this more..

Reply:
hey @Rob--W. I tried to skip this action per the undo/redo instructions but it wasnt working as expected so I adjusted the userEdit value for this and it *seems* to be ok. But this is what we want - to remove the image from local storage.